### PR TITLE
fix: Handling padding present in some devices.

### PR DESCRIPTION
**Pull request by panmari**
## Description

Using width directly doesn't account for the additional padding that is
returned by some cameras. For further references, see e.g.
https://gist.github.com/Alby-o/fe87e35bc21d534c8220aed7df028e03#gistcomment-3218349

Image labeller started detecting real labels on my test devices (Pixel
3a and Pixel3) with this patch.

## Related Issues

* https://github.com/FirebaseExtended/flutterfire/issues/5501
* https://github.com/FirebaseExtended/flutterfire/issues/3798

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by panmari** [comment]
Bump, can somebody please take a look? Do you need some other type of confirmation that HEAD doesn't work on some devices?

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionHandler.java [hunk 1]
```diff
@@ -131,10 +132,16 @@ private FirebaseVisionImage dataToVisionImage(Map<String, Object> imageData) thr
       case "bytes":
         @SuppressWarnings("unchecked")
         Map<String, Object> metadataData = (Map<String, Object>) imageData.get("metadata");
-
+        ArrayList<Map<String, Object>> planeMetadata =
```
Comment (rrousselGit) [inline packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionHandler.java:135]: Could you add tests for this?

File: packages/firebase_ml_vision/pubspec.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: firebase_ml_vision
 description: Flutter plugin for Firebase machine learning vision services.
 homepage: https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_ml_vision
-version: 0.11.0+2
+version: 0.12.0
```
Comment (rrousselGit) [inline packages/firebase_ml_vision/pubspec.yaml:4]: Versions are updated on release, could you revert this?

Comment (panmari) [reply packages/firebase_ml_vision/pubspec.yaml]: Done.

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionHandler.java (hunk 1 above)
Comment (panmari) [reply packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionHandler.java:135]: What type of test would you suggest? I looked into the flutter driver test, but mocking the necessary CameraImage seems hard.

Comment (panmari) [reply packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionHandler.java:135]: Gentle ping, any advice on this?

**Comment by Salakar** [comment]
I'm not sure if we'll be able to merge this as the `firebase_ml_vision` package is now discontinued since its APIs have been deprecated and removed from the Android & iOS Firebase SDKs.

I'd recommend switching to the alternatives;

 - **for on-device vision APIs**; switch to Google's standalone ML Kit library via [google_ml_kit](https://pub.dev/packages/google_ml_kit)
 - **for cloud vision APIs**; the recommended approach now is to use Firebase
Authentication and Functions, which gives you a managed, serverless gateway to the Google Cloud Vision APIs. For an example Functions project see the [vision-annotate-images](https://github.com/firebase/functions-samples/tree/master/vision-annotate-images) sample project. You'd deploy this project for example and then call the cloud function from your app with your image data to get the annotation result.

**Comment by russellwheatley** [comment]
Closing this PR as `ml_vision` package is now deprecated.

**Pull request closed without merging**